app/_src/kubernetes-ingress-controller/reference/failure-modes.md

```diff
@@ -0,0 +1,88 @@
+
+## Failures in translating configuration
+
+When {{site.kic_product_name}} finds Kubernetes resources that can't be correctly translated to {{site.base_gateway}} configuration (for example, an `Ingress` is using a non-exist `Service` as its backend), a translation failure is generated with the namespace and name of the objects causing the failure. The Kubernetes objects causing translation failures will not be translated to {{site.base_gateway}} configuration in the translation process. You can use Kubernetes events and Prometheus metrics to observe the translation failures. If {{site.kic_product_name}} is integrated with {{site.konnect_product_name}}, it will report that translation error happened in uploading node status.
+
+{{site.kic_product_name}} collects all translation failures and generates a Kubernetes `Event` with the `Warning` type and the `KongConfigurationTranslationFailed` reason per each causing object in a translation failure. Prometheus metrics could also reflect the statistics of translation failures: 
+* `ingress_controller_translation_broken_resource_count` is the number of translation failures that happened in the latest translation
+* `ingress_controller_translation_count` with the `success=false` label is the total number of translation procedures where translation failures happen
+
+You can use `kubectl get events -n <namespace> --field-selector reason="KongConfigurationTranslationFailed"` to fetch events generated for translation failures. For example, if an `Ingress` `ing-1` in namespace `test` used a non-exist `Service` as its backend, you can get the event with the following command:
+
+```bash
+$ kubectl get events -n test --field-selector reason="KongConfigurationTranslationFailed"
+
+LAST SEEN   TYPE      REASON                               OBJECT                    MESSAGE
+18m         Warning   KongConfigurationTranslationFailed   ingress/ing-1   failed to resolve Kubernetes Service for backend: failed to fetch Service test/httpbin-deployment-1: Service test/httpbin-deployment-1 not found
+```
+
+## Failures in applying configuration to {{site.base_gateway}}
+
+When {{site.kic_product_name}} fails to apply translated {{site.base_gateway}} configuration to {{site.base_gateway}}, {{site.kic_product_name}} will try to recover from the failure, and also record the failure by logs, Kubernetes events and Prometheus metrics. This usually fails because the translated configuration is rejected by {{site.base_gateway}}.
```

Comment:
```suggestion
When {{site.kic_product_name}} fails to apply translated {{site.base_gateway}} configuration to {{site.base_gateway}}, {{site.kic_product_name}} will try to recover from the failure and record the failure into logs, Kubernetes events, and Prometheus metrics. 
Recovery usually fails because the translated configuration is rejected by {{site.base_gateway}}.
```
Is this right - is recovery what's failing?